bika/lims/tests/test_limitdetections.py

```diff
@@ -366,10 +365,13 @@ def test_ar_manageresults_limitdetections(self):
         ar = create_analysisrequest(client, request, values, services)
 
         # Basic detection limits
-        asidxs = {'analysisservice-3': 0,
-                  'analysisservice-6': 1,
-                  'analysisservice-7': 2}
-        for a in ar.getAnalyses():
+        asidxs = {'Ca': 0, # analysisservice-3
+                  'Cu': 1, # analysisservice-6
+                  'Fe': 2} # analysisservice-7
+        ans = ar.getAnalyses()
+        # Sort them by getKeyword, so we get them in the same order as the ASs
+        ans = sorted(ans, key=lambda x: x.getId)
```

Comment:
You're sorting her by `getId` and not by `getKeyword` as described by the comment